vm-image-spec.yaml

```diff
@@ -312,6 +312,12 @@ files:
         query: |
           SELECT checkpoints_timed FROM pg_stat_bgwriter;
 
+      - metric_name: logical_snapshot_files
+        type: guage
+        help: 'Number of snapshot files in pg_logical/snapshot'
+        values: [num_files]
+        query: SELECT COUNT(*) AS num_files FROM pg_ls_logicalsnapdir() WHERE name LIKE '%.snap';
```

Comment:
Two questions here
1. Does Postgres store anything else in addition to the `*.snap` files in the same directory?
2. Does our aux files persistence logic looks exactly at `*.snap` or it just scrapes everything within this directory?

Reply:
1. During the building of snapshots, postgres will create temporary files that get renamed to the final filename at the end of it all. We do not WAL log this file[^1], only the actual snap file.
2. Pageserver looks for XLOG_LOGICAL_MESSAGE records with the prefix `neon-file:`[^2]. Then it adds the files to the timeline as an aux file.

[^1]: https://github.com/neondatabase/postgres/blob/47a9122a5a150a3217fafd3f3d4fe8e020ea718a/src/backend/replication/logical/snapbuild.c#L1759
[^2]: https://github.com/neondatabase/neon/blob/fa0750a37e01cee2e909d91be9b556ee2f128406/pageserver/src/walingest.rs#L381-L398